src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPool.cs

```diff
@@ -21,9 +21,16 @@ public HttpConnectionPool()
 
         public HttpConnection GetConnection()
         {
-            HttpConnection connection;
-            if (_idleConnections.TryTake(out connection))
+            while (_idleConnections.TryTake(out HttpConnection connection))
             {
+                if (connection.InitialReadCompleted)
+                {
+                    // We got a connection, but it was either already closed by the server or the
```

Comment:
Makes sense.

We need some sort of client timeout for idle connections anyway.  As part of that logic we can potentially check for server disconnect as well.  Won't catch it right away, but that seems fine.